PhysicsTools/NanoAOD/plugins/IsolatedTrackCleaner.cc

```diff
@@ -0,0 +1,67 @@
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/Candidate/interface/Candidate.h"
+#include "DataFormats/PatCandidates/interface/IsolatedTrack.h"
+#include "DataFormats/Common/interface/RefToPtr.h"
+#include "CommonTools/Utils/interface/StringCutObjectSelector.h"
+
+#include <vector>
+
+
+class IsolatedTrackCleaner : public edm::global::EDProducer<> {
+    public:
+        IsolatedTrackCleaner( edm::ParameterSet const & params ) :
+            tracks_(consumes<std::vector<pat::IsolatedTrack>>(params.getParameter<edm::InputTag>("tracks"))),
+            cut_(params.getParameter<std::string>("cut"))
+        {
+            for (const edm::InputTag & tag : params.getParameter<std::vector<edm::InputTag>>("finalLeptons")) {
+                leptons_.push_back(consumes<reco::CandidateView>(tag));
+            }
+            produces<std::vector<pat::IsolatedTrack>>();
```

Comment:
given no Nano-specific outputs, this should probably be moved to a more common place

Reply:
will move to PatAlgos. It may make sense (in a separate PR) to move some stuff that is more related to miniAOD than to PAT into to e.g. PhysicsTools/MiniAODProducers